### ISSUE TITLE
API Renderer Map not populated in certain load orders

### ISSUE THREAD

**Issue by cech12** [Tanticle/baubles#1]
I am the developer of the Useful Hats mod and I am trying to add support for your Baubles 2 mod. 

It works really good except for the rendering of my hats:

![grafik](https://github.com/Tanticle/baubles/assets/16005054/83206420-b439-4b6f-876c-2cd795664d94)

As far as I can see, the translation to the head is not done before the `renderHead` method is called. And I cannot do it by myself, because the following parameters are missing: `limbAngle`, `limbDistance`, `animationProgress`, `headYaw` and `headPitch`.

Trinkets has a helper method for that, which I am using to translate to the head position. 
https://github.com/emilyploszaj/trinkets/blob/main/src/main/java/dev/emi/trinkets/api/client/TrinketRenderer.java#L40

But when I copy this method and use it in the Baubles renderer, the model rotates too wide:

![grafik](https://github.com/Tanticle/baubles/assets/16005054/d0f48a75-1220-47fb-b6ca-bc894d749cce)

Another thing is, that I cannot use `BaublesAPI.getRenderers()` because it is `null` during the `FMLClientSetupEvent`. Using `BaublesClient.RENDERERS` works for me right now.

I am testing with the version: Pre-Release 0.4

Until this is fixed, I disable the rendering in the Baubles slots for my mod :)
Thanks for your help :)

**Comment by Tanticle** [comment]
Can you show me your implementation of the renderer? I need to see what exactly you're doing.

**Comment by Tanticle** [comment]
I found your code. It looks to me like you are applying additional transformations to the render pose using the `followBodyRotations` method in your `AbstractUsefulHatsRenderer`. The `renderHead` method is already translated and rotated to the values of the head model part, and by copying the rotation from the biped model again, you are offsetting the rotation from the actual position and rotation twice. Your hat model does not need any rotation or offsets, other than the offset from the center of the bottom face of the head cuboid to wherever the first vertex is drawn.

**Comment by cech12** [comment]
That should do the trick ^^ Yesterday it was late for me xD I will look into that when I have time. Thanks for reviewing my code :)

**Comment by Tanticle** [comment]
If you check the source, there is an example source set with a test ring and its own renderer. You can take a look at that for a reference of what you need to do yourself. As for the getRenderer thing, I will have to look into that, it looks like the init event fires before the Baubles mod actually gets to set it.

**Comment by cech12** [comment]
I am using the `HumanoidModel` and the included `renderToBuffer` method which also is translating the model to a specific position. I fixed the rendering issue by popping the head translation pose from the pose stack and after rendering I pushed a new empty pose. That worked for me^^
https://github.com/cech12/UsefulHats/commit/74946a5979cbb83275d7d011f902d8a3f84e6301#diff-f19fea2c1a507942e9a85c5f6874f4b7de6c62512ca3ae318fd8f58e977a3530

I let the issue open for the getRenderer issue. :)
